src/main.cpp

```diff
@@ -6205,14 +6277,22 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         vRecv >> filter;
     
         if (!filter.IsWithinSizeConstraints())
+          {
             // There is no excuse for sending a too-large filter
             Misbehaving(pfrom->GetId(), 100);
+            return false;
+          }
         else
         {
             LOCK(pfrom->cs_filter);
             delete pfrom->pfilter;
             pfrom->pfilter = new CBloomFilter(filter);
-            pfrom->pfilter->UpdateEmptyFull();
+            if (!pfrom->pfilter)
```

Comment:
I was thinking out of memory on non virtual memory machines like many ARM.  I was considering memory exhaustion attacks.  Yes, to make this change really work we'd need some code that does some cleanup actions, like disconnecting a bunch of nodes, and preferably the CNode* that is leaking memory.  

However, serving filtered tx to SPV clients may not be the key purpose of this node so I here I am suggesting that we degrade gracefully in this situation, and let the next mission-critical allocation cause the node to go down.